File: baselib/src/main/java/com/yyc/baselib/utils/CreatUtil.java
package com.yyc.baselib.utils;

import java.lang.reflect.ParameterizedType;

/**
 * Author: Austin
 * Time: 2018/7/18
 * Description:
 */

public class CreatUtil {
    /**
     * 获取T类型然后自动new出对象
     * @param o
     * @param i
     * @param <T>
     * @return
     */
    public static <T> T getT(Object o, int i) {

        try {
            return ((Class<T>) ((ParameterizedType) (o.getClass().getGenericSuperclass())).getActualTypeArguments()[i]).newInstance();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;

    }
}

File: baselib/src/main/java/com/yyc/baselib/utils/CreatUtil.kt
package com.yyc.baselib.utils

import java.lang.reflect.ParameterizedType

/**
 * Author: Austin
 * Time: 2018/7/18
 * Description:
 */

object CreatUtil {
    /**
     * 获取T类型然后自动new出对象
     * @param o
     * @param i
     * @param <T>
     * @return
    </T> */
    fun <T> getT(o: Any, i: Int): T? {

        try {
            return ((o.javaClass.genericSuperclass as ParameterizedType).actualTypeArguments[i] as Class<T>).newInstance()
        } catch (e: Exception) {
            e.printStackTrace()
        }

        return null

    }
}
